### PR TITLE
Echo-server ALB compatibility

### PR DESCRIPTION
**Pull request by dylanbannon**
## what
* Adding the ability to choose between NGiNX and ALB backends for the `echo-server` component.

## why
* Sometimes, clusters use ALB instead of NGiNX.

### CODE REVIEW

**Comment by dylanbannon** [comment]
@Nuru can you confirm this works with ALB?

**Review by max-lobur — APPROVED** [review]

File: modules/echo-server/charts/echo-server/templates/ingress.yaml [hunk 1]
```diff
@@ -1,55 +1,55 @@
...
+    kubernetes.io/ingress.class: {{ .Values.ingress.nginx.class }}
+      {{- if (index .Values.ingress.nginx "tls_certificate_cluster_issuer") }}
+    cert-manager.io/cluster-issuer: {{ .Values.ingress.nginx.tls_certificate_cluster_issuer }}
+      {{- end }}
+    {{- else if eq (printf "%v" .Values.ingress.alb.enabled) "true" }}
+    kubernetes.io/ingress.class: {{ .Values.ingress.alb.class }}
+    alb.ingress.kubernetes.io/load-balancer-name: {{ index .Values.ingress.alb "load_balancer_name" | default "k8s-common" }}
+    alb.ingress.kubernetes.io/group.name: {{ index .Values.ingress.alb "group_name" | default "common" }}
+    alb.ingress.kubernetes.io/scheme: internet-facing
+    alb.ingress.kubernetes.io/load-balancer-attributes: access_logs.s3.enabled=false
```
Comment (max-lobur) [inline modules/echo-server/charts/echo-server/templates/ingress.yaml:25]: This kind of annotations need to be moved to a "default" alb created within the alb controller component. After that removed from here. They are not allowed to be changed per ingress, they are global to an ALB (global among users of a single alb.ingress.kubernetes.io/load-balancer-name).

Comment (aknysh) [reply modules/echo-server/charts/echo-server/templates/ingress.yaml:21]: besides this comment, everything else looks good

**Review by aknysh — COMMENTED** [review]
please see comments

File: modules/echo-server/charts/echo-server/templates/ingress.yaml (hunk 1 above)
Comment (Nuru) [reply modules/echo-server/charts/echo-server/templates/ingress.yaml:21]: @max-lobur @aknysh We do not have a standard mechanism for parallel configuration of Ingresses in that way, and are not about to develop one today for the open source component. Can the `load-balancer-attributes` annotation be safely left off? If not, I say we stick with this and require `echo-server` to use its own Load Balancer if needed.

**Comment by Nuru** [comment]
Ignore the pre-commit check failure: it is complaining that the template in the Helm chart are not valid Kubernetes resources (which of course they are not, they are Go templates).

**Review by Nuru — CHANGES_REQUESTED** [review]
Please verify this still works with Nginx after my changes

**Review by Nuru — APPROVED** [review]

File: modules/echo-server/charts/echo-server/templates/ingress.yaml (hunk 1 above)
Comment (max-lobur) [reply modules/echo-server/charts/echo-server/templates/ingress.yaml]: Agreed. Securing annotations should be a separate PR

**Pull request merged**